src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAISerializationHelpers.cs

```diff
@@ -0,0 +1,100 @@
+    }
+
+    /// <summary>
+    /// Serializes a Microsoft.Extensions.AI completion using the OpenAI wire format.
+    /// </summary>
+    /// <param name="stream">The stream to write the value.</param>
+    /// <param name="chatCompletion">The chat completion to serialize.</param>
+    /// <param name="options">The <see cref="JsonSerializerOptions"/> governing function call content serialization.</param>
+    /// <param name="cancellationToken">A token used to cancel the serialization operation.</param>
+    /// <returns>A task tracking the serialization operation.</returns>
+    public static async Task SerializeAsync(
+        Stream stream,
+        ChatCompletion chatCompletion,
+        JsonSerializerOptions? options = null,
+        CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(stream);
+        _ = Throw.IfNull(chatCompletion);
+        options ??= AIJsonUtilities.DefaultOptions;
+
+        OpenAI.Chat.ChatCompletion openAiChatCompletion = OpenAIModelMappers.ToOpenAIChatCompletion(chatCompletion, options);
+        BinaryData binaryData = JsonModelHelpers.Serialize(openAiChatCompletion);
+#if NET
+        await stream.WriteAsync(binaryData.ToMemory(), cancellationToken).ConfigureAwait(false);
+#else
+        await stream.WriteAsync(binaryData.ToArray(), 0, binaryData.Length, cancellationToken).ConfigureAwait(false);
+#endif
+    }
+
+    /// <summary>
+    /// Serializes a Microsoft.Extensions.AI streaming completion using the OpenAI wire format.
+    /// </summary>
+    /// <param name="stream">The stream to write the value.</param>
+    /// <param name="streamingChatCompletionUpdates">The streaming chat completions to serialize.</param>
+    /// <param name="options">The <see cref="JsonSerializerOptions"/> governing function call content serialization.</param>
+    /// <param name="cancellationToken">A token used to cancel the serialization operation.</param>
+    /// <returns>A task tracking the serialization operation.</returns>
+    public static Task SerializeAsSseEventsAsync(
+        Stream stream,
+        IAsyncEnumerable<StreamingChatCompletionUpdate> streamingChatCompletionUpdates,
+        JsonSerializerOptions? options = null,
+        CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(stream);
+        _ = Throw.IfNull(streamingChatCompletionUpdates);
+        options ??= AIJsonUtilities.DefaultOptions;
+
+        var mappedUpdates = OpenAIModelMappers.ToOpenAIStreamingChatCompletionAsync(streamingChatCompletionUpdates, options, cancellationToken);
+        return SseFormatter.WriteAsync(ToSseEventsAsync(mappedUpdates), stream, FormatAsSseEvent, cancellationToken);
+
+        static async IAsyncEnumerable<SseItem<BinaryData>> ToSseEventsAsync(IAsyncEnumerable<OpenAI.Chat.StreamingChatCompletionUpdate> updates)
+        {
+            await foreach (var update in updates.ConfigureAwait(false))
```

Comment:
Should this be factoring in the cancellation token?

Reply:
I'm not sure that it's necessary. It's being passed both in the enumerator that it wraps and the formatter that consumes this IAE.